purchase_warn_message/models/purchase_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    purchase_warn_msg = fields.Text(compute="_compute_purchase_warn_msg")
+
+    def _compute_purchase_warn_msg(self):
+        for rec in self:
+            if rec.partner_id:
+                if rec.partner_id.parent_id:
+                    p = rec.partner_id.parent_id
+                else:
+                    p = rec.partner_id
+                if rec.state not in ["done", "cancel"] and p.purchase_warn == "warning":
+                    rec.purchase_warn_msg = p.purchase_warn_msg
+                    continue
```

Comment:
why do you use continue?
you can add a variable after the `if rec.partner_id:` and only change the value in this if, and only use one write in the function. `rec.purchase_warn_msg = new_variable`.